### PR TITLE
test build with 18 

### PR DESCRIPTION
**Pull request by olamy**
Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MCOMPILER) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[MCOMPILER-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MCOMPILER-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [ ] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [ ] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

File: .github/workflows/maven-verify.yml [hunk 1]
```diff
@@ -27,4 +27,5 @@ jobs:
     uses: apache/maven-gh-actions-shared/.github/workflows/maven-verify.yml@v2
     with:
       jdk-distribution-matrix: '[ "temurin", "zulu", "microsoft", "liberica", "adopt-openj9" ]'
```
Comment (mthmulders) [inline .github/workflows/maven-verify.yml:29]: I know it's not part of this change, but do we really expect big differences between let's say Liberia, Zulu, Microsoft and Temurin? I do see why OpenJ9 is on the list, but all the others are regular Hotspot JVM's, aren't they?

Comment (olamy) [reply .github/workflows/maven-verify.yml:29]: probably not :) but as I already got differences (for javadoc) between distributions

**Review by slachiewicz — APPROVED** [review]

File: .github/workflows/maven-verify.yml (hunk 1 above)
Comment (olamy) [reply .github/workflows/maven-verify.yml:29]: @mthmulders what would be your suggestion to limit to which?

Comment (mthmulders) [reply .github/workflows/maven-verify.yml:29]: Usually, I'd say "a Hotspot VM and an OpenJ9 VM". For OpenJ9, we don't have my choices; for Hotspot, I guess Temurin or maybe Zulu?

But if you're saying there are differences (e.g. in generated Javadoc) and they matter for the m-c-p, then we should include them.

File: .github/workflows/maven-verify.yml [hunk 2]
```diff
@@ -24,7 +24,10 @@ on:
 jobs:
   build:
     name: Verify
-    uses: apache/maven-gh-actions-shared/.github/workflows/maven-verify.yml@v2
+    uses: apache/maven-gh-actions-shared/.github/workflows/maven-verify.yml@v3
     with:
-      jdk-distribution-matrix: '[ "temurin", "zulu", "microsoft", "liberica", "adopt-openj9" ]'
-      matrix-exclude: '[{ "jdk": "8", "distribution": "microsoft"}]'
+      maven-args: "-Dinvoker.parallelThreads=2"
```
Comment (slawekjaranowski) [inline .github/workflows/maven-verify.yml:29]: Why not in project configuration?
I like values as 1C, 2C, ...

Comment (olamy) [reply .github/workflows/maven-verify.yml:29]: because need some good environment and might not suit everyone.

Comment (olamy) [reply .github/workflows/maven-verify.yml]: btw doesn't work (e.g concurrent access failure https://github.com/apache/maven-compiler-plugin/runs/7444447535?check_suite_focus=true)

Comment (slawekjaranowski) [reply .github/workflows/maven-verify.yml]: looks like problem in Maven itself

```
Caused by: java.io.FileNotFoundException: .../local-repo/org/sonatype/plexus/plexus-build-api/0.0.4/plexus-build-api-0.0.4.pom.part (No such file or directory)
``

Comment (olamy) [reply .github/workflows/maven-verify.yml]: yup concurrent download of artifact

**Pull request merged**